python/ray/tune/utils/callback.py

```diff
@@ -3,22 +3,99 @@
 
 from ray.tune.callback import Callback
 from ray.tune.syncer import SyncConfig
+from ray.tune.logger import CSVLogger, DEFAULT_LOGGERS, ExperimentLogger, \
+    JsonLogger, LegacyExperimentLogger, Logger
 from ray.tune.syncer import SyncerCallback
 
 
 def create_default_callbacks(callbacks: Optional[List[Callback]],
-                             sync_config: SyncConfig):
+                             sync_config: SyncConfig,
+                             loggers: Optional[List[Logger]]):
 
     callbacks = callbacks or []
+    has_syncer_callback = False
+    has_csv_logger = False
+    has_json_logger = False
 
-    # Check if there is a SyncerCallback
-    has_syncer_callback = any(isinstance(c, SyncerCallback) for c in callbacks)
+    # Track syncer obj/index to move callback after loggers
+    last_logger_index = None
+    syncer_index = None
+
+    if not loggers:
+        # If no logger callback and no `loggers` have been provided,
+        # add DEFAULT_LOGGERS.
+        if not any(
+                isinstance(callback, ExperimentLogger)
+                for callback in callbacks):
+            loggers = DEFAULT_LOGGERS
+
+    # Create LegacyExperimentLogger for passed Logger classes
+    if loggers:
+        # Todo(krfricke): Deprecate `loggers` argument, print warning here.
+        add_loggers = []
+        for trial_logger in loggers:
+            if isinstance(trial_logger, ExperimentLogger):
+                callbacks.append(trial_logger)
+            elif isinstance(trial_logger, type) and issubclass(
+                    trial_logger, Logger):
+                add_loggers.append(trial_logger)
+            else:
+                raise ValueError(
+                    f"Invalid value passed to `loggers` argument of "
+                    f"`tune.run()`: {trial_logger}")
+        if add_loggers:
+            callbacks.append(LegacyExperimentLogger(add_loggers))
+
+    # Check if we have a CSV and JSON logger
+    for i, callback in enumerate(callbacks):
+        if isinstance(callback, LegacyExperimentLogger):
+            last_logger_index = i
+            if CSVLogger in callback.logger_classes:
+                has_csv_logger = True
+            if JsonLogger in callback.logger_classes:
+                has_json_logger = True
+        # Todo(krfricke): add checks for new ExperimentLogger classes
```

Comment:
will this be part of this PR?

Reply:
No, this will be a follow up once we implemented a "native" CSV/JSON ExperimentLogger